addons/product_extended/wizard/wizard_price.py

```diff
@@ -1,60 +1,33 @@
-# -*- encoding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Management Solution	
-#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
-#    Copyright (C) 2010-2011 OpenERP S.A. (<http://www.openerp.com>).
-#    $Id$
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU General Public License as published by
-#    the Free Software Foundation, either version 3 of the License, or
-#    (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU General Public License for more details.
-#
-#    You should have received a copy of the GNU General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# -*- coding: utf-8 -*-
 
 from openerp.exceptions import UserError
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import api, models, fields, _
 
-class wizard_price(osv.osv):
+class WizardPrice(models.Model):
     _name = "wizard.price"
     _description = "Compute price wizard"
-    _columns = {
-        'info_field': fields.text('Info', readonly=True), 
-        'real_time_accounting': fields.boolean("Generate accounting entries when real-time"),
-        'recursive': fields.boolean("Change prices of child BoMs too"),
-        }
 
-    def default_get(self, cr, uid, fields, context=None):
-        res = super(wizard_price, self).default_get(cr, uid, fields, context=context)
-        product_pool = self.pool.get('product.template')
-        product_obj = product_pool.browse(cr, uid, context.get('active_id', False))
-        if context is None:
-            context = {}
-        rec_id = context and context.get('active_id', False)
+    info_field = fields.Text("Info", readonly=True)
+    real_time_accounting = fields.Boolean("Generate accounting entries when real-time")
+    recursive = fields.Boolean("Change prices of child BoMs too")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(WizardPrice, self).default_get(fields)
+        #Product = self.env['product.template']
+        Product = self.env['product.template'].browse(self._context.get('active_id', False))
+        rec_id = self._context and self._context.get('active_id', False)
```

Comment:
1. use `self.env.context`
2. there is always a context in the new API
3. assert that the id is present before using it to browse a model, that generates an inconsistent cache entry
4. no need for the explicit `False`